### PR TITLE
[UI] Show execution details in Run Details Page ML Metadata tab of steps

### PR DESCRIPTION
**Pull request by Bobgy**
Mitigates https://github.com/kubeflow/pipelines/issues/3424 and https://github.com/kubeflow/pipelines/issues/3226

Demo: https://youtu.be/ARsCElYSqq8

Changes:
* Find corresponding ML metadata for pipeline run (context) and steps (execution) from Run Details page
* Refactor execution details page component to be portable
* Show execution details on Run Details' step ML Metadata tab

Current Limitation:
* ML Metadata from mlmd writer doesn't record pod name in execution custom property `kfp_pod_name`, so it isn't recorded at all. Therefore, this only supports tfx. (but we can support kfp dsl later too)

/area frontend

### CODE REVIEW

File: frontend/src/lib/MlmdUtils.ts [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+  } catch (err) {
+    err.message = `Cannot find context with ${JSON.stringify(request.toObject())}: ` + err.message;
+    throw err;
+  }
+}
+
+/**
+ * @throws error when network error, or not found
+ */
+export async function getTfxRunContext(argoWorkflowName: string): Promise<Context> {
```
Comment (numerology) [inline frontend/src/lib/MlmdUtils.ts:34]: Actually this remind me of one thing. See https://github.com/tensorflow/tfx/blob/5ee3de9c77a92e904010c7656d6269d3b28744bf/tfx/orchestration/kubeflow/base_component.py#L136

Users can actually override the TFX run id to be something different than Argo workflow ID, which might cause problem to this piece of logic. 

@neuromage Not sure how useful that customization is currently. Shall we consider closing the door of overriding run_id?

Comment (Bobgy) [reply frontend/src/lib/MlmdUtils.ts:34]: Thanks! that's a good catch.
From the comment
```
# Allow overriding pipeline's run_id externally, primarily for testing.
```
looks like it was for testing.

**Comment by Bobgy** [comment]
/approve
The PR is fairly complete.
@numerology @neuromage You can lgtm if concerns cleared.

**Comment by numerology** [comment]
Thanks! @Bobgy

File: frontend/src/lib/MlmdUtils.ts (hunk 1 above)
Comment (Ark-kun) [reply frontend/src/lib/MlmdUtils.ts:34]: > Actually this remind me of one thing. See https://github.com/tensorflow/tfx/blob/5ee3de9c77a92e904010c7656d6269d3b28744bf/tfx/orchestration/kubeflow/base_component.py#L136
> 
> Users can actually override the TFX run id to be something different than Argo workflow ID, which might cause problem to this piece of logic.
> 
> @neuromage Not sure how useful that customization is currently. Shall we consider closing the door of overriding run_id?

Can we add something explicit like `workflow_name` that's not overridable?

Comment (numerology) [reply frontend/src/lib/MlmdUtils.ts:34]: > Can we add something explicit like workflow_name that's not overridable?

Do you mean here or on the TFX side?

**Comment by Ark-kun** [comment]
> * ML Metadata from mlmd writer doesn't record pod name in execution custom property `kfp_pod_name`, so it isn't recorded at all. Therefore, this only supports tfx. (but we can support kfp dsl later too)

Adding this in https://github.com/kubeflow/pipelines/pull/3479

File: frontend/src/lib/MlmdUtils.ts (hunk 1 above)
Comment (Ark-kun) [reply frontend/src/lib/MlmdUtils.ts:34]: On the TFX side.

Comment (numerology) [reply frontend/src/lib/MlmdUtils.ts:34]: > On the TFX side.

Yes. I think I'll make a CL closing the door.

**Pull request merged**